### PR TITLE
references: update DM FAQ for 4GB binlog related errors

### PR DESCRIPTION
**Pull request by dcalvin**
This PR  updates FAQ to handle the errors resulted by larger binlog file (> 4GB)

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1924
### Which version does your change affect? 

TiDB Dev/3.0/2/1

### CODE REVIEW

File: dev/faq/data-migration.md [hunk 1]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
 
 When this type of error occurs in the current version, use `stop-task` to stop the task and then use `start-task` to restart the task. The automatic retry mechanism of DM will be improved later.
 
-## What can I do when a replication task is interrupted with failing to get or parse binlog errors like `get binlog error ERROR 1236 (HY000)` and `binlog checksum mismatch, data may be corrupted` returned?
+## What can I do when the relay unit throws error `event from * in * diff from passed-in event *` or a replication task is interrupted with failing to get or parse binlog errors like `get binlog error ERROR 1236 (HY000)` and `binlog checksum mismatch, data may be corrupted` returned?
 
-During the DM incremental replication process, this error might occur if the binlog file in the upstream exceeds **4 GB**, and DM encounters a replication interruption when processing this binlog file (including interruption caused by anomalies in ordinary pause or stop tasks).
+During the DM process of relay log pulling or incremental replication, this two errors might occur if the binlog file in the upstream exceeds **4 GB**.
```
Comment (lilin90) [inline dev/faq/data-migration.md]:
```suggestion
During the DM process of relay log pulling or incremental replication, these two errors might occur if the size of the upstream binlog file exceeds **4 GB**.
```

File: dev/faq/data-migration.md [hunk 2]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
...
 
-**Cause:** DM needs to store the replicated binlog position, and MySQL officially uses `uint32` to store it, so the binlog position of the file with offset exceeding **4 GB** overflows and an incorrect binlog position is stored. After the task or DM-worker is restarted, this incorrect binlog position is used to re-parse the binlog or relay log.
+**Cause:** When writing relay logs, DM needs to perform event verification based on binlog positions and the size of the binlog file, and store the replicated binlog positions as checkpoints. However, the official method of MySQL to store binlog positions is `uint32`. This means the binlog position offset for a binlog file over 4 GB overflows, hence the errors above.
```
Comment (lilin90) [inline dev/faq/data-migration.md]: `hence` is usually used as an adverb.

```suggestion
**Cause:** When writing relay logs, DM needs to perform event verification based on binlog positions and the size of the binlog file, and store the replicated binlog positions as checkpoints. However, the official MySQL uses `uint32` to store binlog positions. This means the offset value of binlog positions over 4 GB overflows and then the errors above occur.
```

File: dev/faq/data-migration.md [hunk 3]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
...
 
-In this case, manually recover replication using the following solution:
+For relay units, manually recover replication using the following solution:
 
-1. Determine whether the error occurs during the write of the relay log or replication of Binlog replication/Syncer unit (according to the component information in the log error message).
+1. Identify in the upstream that the corresponding binlog file has exceeded 4GB when the error occurs.
```
Comment (lilin90) [inline dev/faq/data-migration.md]:
```suggestion
1. Identify in the upstream that the corresponding binlog file size has exceeded 4GB when the error occurs.
```

File: dev/faq/data-migration.md [hunk 4]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
...
+
+4. In the relay log directory, update the corresponding `relay.meta` file to pull from the next binlog position.
+
+    Example: when the error occurs, `binlog-name = "mysql-bin.004451"` and `binlog-pos = 2453`. Update them respectively to `binlog-name = "mysql-bin.004452"` and `binlog-pos = 4`.
+
+5. Restart the DM-worker.
+
+For binlog replication processing units, manually recover replication using the following solution:
+
+1. Identify in the upstream the corresponding binlog file has exceeded 4GB when error occurs.
```
Comment (lilin90) [inline dev/faq/data-migration.md]:
```suggestion
1. Identify in the upstream that the corresponding binlog file size has exceeded 4GB when the error occurs.
```

File: dev/faq/data-migration.md [hunk 5]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
...
+
+2. Stop the replication task using `stop-task`.
+
+3. Update the `binlog_name` in the global checkpoints of the downstream  `dm_meta` database and in each table to the name of the binlog file in error; update `binlog_pos` to a valid position value for which replication has completed, for example, 4.
```
Comment (lilin90) [inline dev/faq/data-migration.md]:
```suggestion
3. Update the `binlog_name` in the global checkpoints of the downstream `dm_meta` database and in each table checkpoint to the name of the binlog file in error; update `binlog_pos` to a valid position value for which replication has completed, for example, 4.
```

File: dev/faq/data-migration.md [hunk 6]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
...
+
+    Example: the name of the task in error is `dm_test`, the corresponding s`source-id` is `replica-1`, and the corresponding binlog file is `mysql-bin|000001.004451`. Execute the following:
```
Comment (lilin90) [inline dev/faq/data-migration.md]:
```suggestion
    Example: the name of the task in error is `dm_test`, the corresponding `source-id` is `replica-1`, and the corresponding binlog file is `mysql-bin|000001.004451`. Execute the following command:
```

File: dev/faq/data-migration.md [hunk 7]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
...
+
+    ```sql
+    `UPDATE dm_test_syncer_checkpoint SET binlog_name='mysql-bin|000001.004451', binlog_pos = 4 WHERE id='replica-1';`
```
Comment (lilin90) [inline dev/faq/data-migration.md]: Remove extra inline code format.

```suggestion
    UPDATE dm_test_syncer_checkpoint SET binlog_name='mysql-bin|000001.004451', binlog_pos = 4 WHERE id='replica-1';
```

File: dev/faq/data-migration.md [hunk 8]
```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
...
+
+1. Identify in the upstream the corresponding binlog file has exceeded 4GB when error occurs.
+
+2. Stop the replication task using `stop-task`.
+
+3. Update the `binlog_name` in the global checkpoints of the downstream  `dm_meta` database and in each table to the name of the binlog file in error; update `binlog_pos` to a valid position value for which replication has completed, for example, 4.
+
+    Example: the name of the task in error is `dm_test`, the corresponding s`source-id` is `replica-1`, and the corresponding binlog file is `mysql-bin|000001.004451`. Execute the following:
+
+    ```sql
```
Comment (lilin90) [inline dev/faq/data-migration.md]:
```suggestion
    {{< copyable "sql" >}}
    
    ```sql
```

File: dev/faq/data-migration.md (hunk 2 above)
Comment (dcalvin) [reply dev/faq/data-migration.md]: > binlog positions over 4 GB

Can you confirm @zimulala

Comment (csuzhangxc) [reply dev/faq/data-migration.md]: I think `This means the binlog position for a binlog file over 4 GB overflows, hence the errors above.` is fine. @dcalvin @lilin90

File: dev/faq/data-migration.md (hunk 5 above)
Comment (csuzhangxc) [inline dev/faq/data-migration.md]:
```suggestion
3. Update the `binlog_name` in the global checkpoints and in each table checkpoint of the downstream `dm_meta` database to the name of the binlog file in error; update `binlog_pos` to a valid position value for which replication has completed, for example, 4.
```

File: dev/faq/data-migration.md (hunk 2 above)
Comment (dcalvin) [reply dev/faq/data-migration.md]: Thanks for confirming

**Comment by lilin90** [comment]
@dcalvin Please fix the CI failure.

**Comment by dcalvin** [comment]
> @dcalvin Please fix the CI failure.

Done.

**Pull request merged**